nav2_controller/src/controller_server.cpp

```diff
@@ -746,6 +748,13 @@ void ControllerServer::publishZeroVelocity()
   velocity.header.frame_id = costmap_ros_->getBaseFrameID();
   velocity.header.stamp = now();
   publishVelocity(velocity);
+}
+
+void ControllerServer::onGoalExit()
+{
+  if (publish_zero_velocity_) {
+    publishZeroVelocity();
+  }
```

Comment:
I have 1 question about this option.
What would happend if the goal_checker kick in when we have such a option ?
As far as I know when we are in the goal_checker tolerance, controller will publish zero velocity. Does this place an impact on that part ?

Reply:
> As far as I know when we are in the goal_checker tolerance, controller will publish zero velocity

That is ideally the case, but not always implemented if someone's plugin is not well thought out. This is a protection mechanism. Its better to publish a zero twice than not at all - especially for robots that don't have in their firmware to stop executing a command after a certain timeout. And trust me, more robot platforms out there than you would think will continue to drive indefinitely with an extraneous command than you think (its a little scary).

Reply:
> controller will publish zero velocity

Only if implemented. This is exactly what we're trying to avoid here by making this behavior optional in the controller server.